packages/ttypescript/src/loadTypescript.ts

```diff
@@ -1,32 +1,66 @@
-import * as fs from 'fs';
-import * as resolve from 'resolve';
 import * as TS from 'typescript';
 type ts = typeof TS;
+import { readFileSync } from 'fs';
+import { sync as resolveSync } from 'resolve';
 import { patchCreateProgram } from './patchCreateProgram';
 import { dirname } from 'path';
 import { runInThisContext } from 'vm'
+import Module = require('module')
 
+type TypeScriptModuleFactory = (exports: ts, require: NodeRequireFunction, module: { exports: any }, filename: string, dirname: string) => ts;
+const typeScriptModuleFactoryCache: { [key: string]: TypeScriptModuleFactory } = Object.create(null)
```

Comment:
module: { exports: ts }

Reply:
Why do we need `typeScriptModuleFactoryCache`? We call `loadTypescript` only once

Reply:
We need because if somebody does `require('ttypescript')` from inside a plugin running tsc we may compile twice typescript.js

Reply:
An option to remove this cache would be to change the way we pass the 'forceLoadConfig' flag, so we can just do require('./typescript') instead of calling loadTypeScript again but I would say that this would deserve another PR.